client/src/app/common/CustomRules/rules-utils.tsx

```diff
@@ -1,25 +1,82 @@
 import { IReadFile, ParsedRule, Rule, Ruleset } from "@app/api/models";
 import yaml from "js-yaml";
+type RuleFileType = "YAML" | "XML" | null;
+export const checkRuleFileType = (filename: string): RuleFileType => {
+  const fileExtension = filename.split(".").pop();
```

Comment:
I wonder if we want to do `filename.toLowerCase().split(".").pop();` here in case someone has capital `.YAML`/`.YML`/`.XML`. Not that that's common